plugins/jobs/web_client/js/views/JobListWidget.js

```diff
@@ -85,6 +129,30 @@ girder.views.jobs_JobListWidget = girder.View.extend({
         window.setTimeout(function () {
             tr.removeClass('g-highlight');
         }, 1000);
+    },
+    _filterJobs: function (jobs) {
+        var filterJobs = [], self = this;
+
+        filterJobs = this.collection.filter(function (job) {
+            return ((_.isEmpty(self.typeFilter) || _.isUndefined(job.attributes.type) ||
+                        self.typeFilter[job.attributes.type]) &&
+                    (_.isEmpty(self.statusFilter) || _.isUndefined(job.attributes.status) ||
+                        self.statusFilter[girder.jobs_JobStatus.text(job.attributes.status)]));
+        });
+
+        return filterJobs;
+    },
+    _updateFilter: function (filter, newValues) {
```

Comment:
Can you provide a comment here explaining why you go to this trouble of updating the existing filter rather than just generating a new one based on the current set of unique values?

I think it relates to maintaining the current state of the filter for any filter states that will remain (were not removed), but I'm not sure.